quickwit-indexing/src/actors/indexing_service.rs

```diff
@@ -293,6 +293,23 @@ impl IndexingService {
         let index_metadata = self.metastore.index_metadata(index_id).await?;
         Ok(index_metadata)
     }
+
+    async fn shutdown_pipeline(
+        &mut self,
+        index_id: String,
+        source_id: String,
+        _ctx: &ActorContext<Self>,
+    ) -> Result<(), IndexingServiceError> {
+        let pipeline_id = IndexingPipelineId {
+            index_id,
+            source_id,
+        };
+        let pipeline_handle_opt = self.pipeline_handles.remove(&pipeline_id);
```

Comment:
Considering how little it does, I'd put that logic directly in the handler.

In particular, the passage from Shutdown `message -> (index_id, source_id) positional argument -> IndexingPipelineId` is offering 2 spots where a bug could be hidden from the compiler and the reader view (positional arguments inverted).

To mitigate, you could also have a
`struct Shutdown(IndexingPipelineId);`